doc/research/nextcloud_integration.md

```diff
@@ -9,45 +11,208 @@
+All files ('add.svg', 'delete.svg', 'map.json', 'tree.png') in all subdirectories ('Icons', 'Maps', 'Pictures') of 'Public' can be accessed.
+The path for fetching a file starts at the shared directory, in this case 'Public'.
+To access the file 'add.svg' following URI is used: 'https://cloud.permaplant.net/public.php/webdav/Icons/add.svg'.
+The first part: 'https://cloud.permaplant.net/' is the scheme and host for our Nextcloud instance.
+The next part '/public.php/webdav/' is the endpoint used to access files from the public share directory specified by the share token.
+The share token is given as username and password in the request.
+The last part 'Icons/add.svg' is the path to the requested file starting from the public share directory.
+
+When we want to fetch 'add.svg' following request is used:
+```bash
+curl -X GET -O -u 2arzyJZYj2oNnHX:2arzyJZYj2oNnHX https://cloud.permaplant.net/public.php/webdav/Icons/add.svg
+```
+
+In PermaplanT we use the public shares for data which should be accessible by unauthenticated users.
+E.g. images on the landing page or icons for plants.
+
+To fetch resources from directories which are not available for the public some form of authentication has to be used.
+The easiest way to authenticate with the Nextcloud instance is to use the username and password in the request. This is called basic authentication.
+However, this is not secure when the requests are sent from the browser as this information can be easily sniffed.
+So we decided to use 'OpenID Connect' for authentication. Details on why we decided to use 'OpenID Connect' can be found in the [decision document]('https://github.com/ElektraInitiative/PermaplanT/blob/master/doc/decisions/auth.md').
+
+All we need to know here is that requests from authenticated users are populated with the access token in the Authorization header.
 
-### testing with cadaver
+Now we can make requests against the '/remote.php/webdav' endpoint.
+This endpoint enables us to fetch files starting from the root directory of the logged in user.
+Given the previously defined directory structure we can now access the file 'secret-doc.txt' with the following URI: 'https://cloud.permaplant.net/remote.php/webdav/Documents/secret-doc.txt'
 
-opening a connection
+### Directory Structure
 
+In the PermaplanT Nextcloud instance we have following directory structure for each user:
 ```bash
-cadaver
-open https://nextcloud.markus-raab.org/nextcloud/remote.php/dav/files/USERNAME/Documents
+PermaplanT/
+└── Maps
+    ├── map_01
+    │   ├── BaseLayer
+    │   │   └── base_layer_image.png
+    │   └── PhotoLayer
+    │       ├── photo1.jpg
+    │       └── photo2.jpg
+    └── map_02
+        ├── BaseLayer
+        │   └── base_layer_image.png
+        └── PhotoLayer
+            ├── photo1.jpg
+            └── photo2.jpg
 ```
 
-type help for a list of commands
+When a map is shared between different users the directory also has to be shared and placed within this structure.
```

Comment:
It is unclear to me what "placed within this structure" means, please elaborate. I think there should be only one "map_01" folder (by the creator of the map) and people with write access get write access to that folder.

Reply:
E.g. If you add me to the shares of a directory of yours (wherever it is in your directory hierarchy), this directory will pop up at my root directory.

Reply:
Ahh, please write this also in the document. I didn't understand because with "sharing map" I didn't think of sharing files in Nextcloud.

I don't know if it helps, but the directory where shares are created can be configured in Settings/Einstellungen -> Sharing/Teilen -> Standardordner für akzeptierte Freigaben wählen

I usually use something like `/Shared/Incoming` to not have a cluttered root directory. But if we move it immediately, it shouldn't matter?

Reply:
The default directory could be useful but I guess moving it to ensure that it is there is the way to go :)